src/parser/simplify.rs

```diff
@@ -296,7 +350,38 @@ impl ResolvableRefs for c::TextOrInlineElement {
 			Reference(e) => sub_res(*e, refs).into(),
 			FootnoteReference(e) => sub_res(*e, refs).into(),
 			CitationReference(e) => sub_res(*e, refs).into(),
-			SubstitutionReference(e) => sub_res(*e, refs).into(),
+			SubstitutionReference(e) => {
+				use crate::document_tree::extra_attributes::ExtraAttributes;
+				if e.extra().refname.len() != 1 {
+					panic!("Expected exactly one name in a substitution reference.");
```

Comment:
Hmm, why is this even a list then?

Reply:
This goes back to docutils treating `refname` attributes as a list. We inherited it from their DTD.